pandas/tests/plotting/test_frame.py

```diff
@@ -3256,6 +3257,117 @@ def test_subplots_sharex_false(self):
         tm.assert_numpy_array_equal(axs[0].get_xticks(), expected_ax1)
         tm.assert_numpy_array_equal(axs[1].get_xticks(), expected_ax2)
 
+    @pytest.mark.parametrize("by", ["C", ["C", "D"]])
+    @pytest.mark.parametrize("column", ["A", ["A", "B"]])
+    def test_hist_plot_by_argument(self, by, column):
+        # GH 15079
+        df = DataFrame(np.random.randn(30, 2), columns=["A", "B"])
+        df["C"] = np.random.choice(["a", "b", "c"], 30)
+        df["D"] = np.random.choice(["a", "b", "c"], 30)
+
+        _check_plot_works(df.plot.hist, column=column, by=by)
+
+    @pytest.mark.slow
+    @pytest.mark.parametrize(
+        "by, column, layout, axes_num",
+        [
+            (["C"], "A", (2, 2), 3),
+            ("C", "A", (2, 2), 3),
+            (["C"], ["A"], (1, 3), 3),
+            ("C", ["A", "B"], (3, 1), 3),
+            (["C", "D"], "A", (9, 1), 9),
+            (["C", "D"], "A", (3, 3), 9),
+            (["C", "D"], ["A"], (5, 2), 9),
+            (["C", "D"], ["A", "B"], (9, 1), 9),
+            (["C", "D"], ["A", "B"], (5, 2), 9),
+        ],
+    )
+    def test_hist_plot_layout_with_by(self, by, column, layout, axes_num):
+        # GH 15079
+        np.random.seed(0)
+        df = DataFrame(np.random.randn(30, 2), columns=["A", "B"])
+        df["C"] = np.random.choice(["a", "b", "c"], 30)
+        df["D"] = np.random.choice(["a", "b", "c"], 30)
+
+        with tm.assert_produces_warning(UserWarning):
```

Comment:
What warning is this throwing?

Reply:
` _check_plot_works` adds an ax so catch warning. see GH #13188, and this is the same as we have in `test_hist_method`